x-pack/plugins/security_solution/public/common/hooks/use_threat_intel_tabs.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { INDICATOR_DESTINATION_PATH } from '../../../common/constants';
+import { TimelineEventsDetailsItem } from '../../../common/search_strategy';
+import { getDataFromSourceHits } from '../../../common/utils/field_formatters';
+import { ThreatDetailsRow, ThreatSummaryRow } from '../components/event_details/helpers';
+import { EventsViewType } from '../components/event_details/event_details';
+
+const sortedThreatSummaryFields = [
+  'matched.field',
+  'matched.type',
+  'provider',
+  'first_seen',
+  'last_seen',
+];
+
+export const useThreatIntelTabs = (
+  data: TimelineEventsDetailsItem[] = [],
+  isAlert: boolean,
+  eventId: string,
+  contextId: string,
+  selectedTabId: EventsViewType
+) => {
+  const threatData: TimelineEventsDetailsItem[][] = useMemo(() => {
+    if (isAlert && data) {
+      const threatIndicator = data.find(
+        ({ field, originalValue }) => field === INDICATOR_DESTINATION_PATH && originalValue
+      );
+      if (!threatIndicator) return [];
+      const { originalValue } = threatIndicator;
+      const values = Array.isArray(originalValue) ? originalValue : [originalValue];
+      return values.reduce((acc, value) => {
+        const item = getDataFromSourceHits(JSON.parse(value));
+        if (item.length > 0) acc.push(item);
+        return acc;
+      }, []);
+    }
+    return [[]];
+  }, [isAlert, data]);
+
+  const threatSummaryRows = useMemo(() => {
+    const rows: ThreatSummaryRow[] = [];
+    if (selectedTabId !== EventsViewType.summaryView) return rows;
+    return threatData
+      .reduce((acc, items) => {
+        items.forEach(({ field, originalValue }) => {
```

Comment:
This `forEach` within the reduce seems indicative of us doing something unnecessary; is it perhaps that we're having to undo part of what `getDataFromSourceHits` does above?

At a high level, this seems like what we need, data-manipulation-wise:

```js
const summaryFields = ['threat.indicator.matched.atomic', 'etc.'];
const indicatorJSON = data.find(field === 'threat.indicator');
const indicatorFields = data.filter(isIndicatorField);
const threatSummaryRows = buildThreatSummaryRows(summaryFields, indicatorFieldsFields);
const threatDetailsRows = buildThreatDetailsRows(indicatorJSON, ALL_INDICATOR_FIELDS).sort('first_seen', 'desc');
```